controllers/dscinitialization/auth.go

```diff
@@ -2,35 +2,94 @@ package dscinitialization
-			Name: serviceApi.AuthInstanceName,
-		},
-		Spec: serviceApi.AuthSpec{
-			AdminGroups:   []string{dashboard.GetAdminGroup()},
-			AllowedGroups: []string{"system:authenticated"},
-		},
-	},
-	)
-	err := r.Create(ctx, defaultAuth)
+func (r *DSCInitializationReconciler) createAuth(ctx context.Context, dscInit *dsciv1.DSCInitialization) error {
+	log := logf.FromContext(ctx)
+	// check for the dashboardConfig kind.
+	// Once the groupsConfig entry in the dashboardConfig is removed this logic can be removed.
+	crd := &apiextv1.CustomResourceDefinition{}
+	err := r.Client.Get(ctx, client.ObjectKey{Name: "odhdashboardconfigs.opendatahub.io"}, crd)
+
+	// can't find the dashboardConfig kind so create the default.
+	if err != nil && k8serr.IsNotFound(err) {
+		log.Info("couldn't find odhDashboardConfig CRD, creating auth with defaults.")
+		err = r.Create(ctx, buildAuth([]string{dashboard.GetAdminGroup()}, []string{"system:authenticated"}))
+		if err != nil && !k8serr.IsAlreadyExists(err) {
+			return err
+		}
+		return nil
+	} else if err != nil {
+		return fmt.Errorf("failed to get odhdashboardconfigs.opendatahub.io CRD: %w", err)
+	}
+
+	// check if a dashboardConfig instance exists.
+	odhObject := &unstructured.Unstructured{}
+	odhObject.SetGroupVersionKind(gvk.OdhDashboardConfig)
+	err = r.Client.Get(ctx, client.ObjectKey{
+		Name:      "odh-dashboard-config",
+		Namespace: dscInit.Spec.ApplicationsNamespace,
+	}, odhObject)
+	if err != nil && k8serr.IsNotFound(err) {
+		log.Info("couldn't find odhDashboardConfig instance, creating auth with defaults.")
+		err = r.Create(ctx, buildAuth([]string{dashboard.GetAdminGroup()}, []string{"system:authenticated"}))
+		if err != nil && !k8serr.IsAlreadyExists(err) {
+			return err
+		}
+		return nil
+	} else if err != nil {
+		return fmt.Errorf("failed to get odh-dashboard-config instance: %w", err)
+	}
+
+	// dashboardConfig CRD exists and we have an instance so copy the groups to the auth CR
+	foundGroups, ok, err := unstructured.NestedStringMap(odhObject.Object, "spec", "groupsConfig")
+	if err != nil {
+		return err
+	}
+	if ok {
+		adminGroup := []string{}
+		allowedGroup := []string{}
+		added := common.AddMissing(&adminGroup, foundGroups["adminGroups"])
+		added += common.AddMissing(&allowedGroup, foundGroups["allowedGroups"])
+
+		// only update if we found a new group in the list
+		if added == 0 {
+			return nil
+		}
+		err = r.Create(ctx, buildAuth(adminGroup, allowedGroup))
+		if err != nil && !k8serr.IsAlreadyExists(err) {
+			return err
+		}
+		return nil
+	}
+
+	// found a dashboardConfig CRD and instance but no groupsConfig or something else strange so just use default
+	err = r.Create(ctx, buildAuth([]string{dashboard.GetAdminGroup()}, []string{"system:authenticated"}))
 	if err != nil && !k8serr.IsAlreadyExists(err) {
 		return err
 	}
 
 	return nil
 }
+
+func buildAuth(adminGroup []string, allowedGroup []string) client.Object {
+	return client.Object(&serviceApi.Auth{
```

Comment:
nit: I think the `client.Object` cast is not needed as `serviceApi.Auth`  implements such interface